### PR TITLE
Template sees parameters leaked from other templates

### LINKED ISSUE #465: Template sees parameters leaked from other templates
I have a resource class declaring/using 2 templates. In each of them, you can see the parameters declared for the other:
Screenshot 2021-12-01 at 12 12 56

Those leaked parameters are then available for completion and are validated as correct.

### PR DESCRIPTION
**Pull request by angelozerr**
Template sees parameters leaked from other templates

Fixes #465

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by angelozerr** [comment]
It should work. I need just to write some tests.

File: qute.jdt/com.redhat.qute.jdt.test/projects/maven/qute-quickstart/src/main/java/org/acme/qute/HelloResource.java [hunk 1]
```diff
@@ -16,6 +16,9 @@
     @Inject
     Template hello;
 
+    @Inject
+    Template goodby;
```
Comment (fbricon) [inline qute.jdt/com.redhat.qute.jdt.test/projects/maven/qute-quickstart/src/main/java/org/acme/qute/HelloResource.java:20]: goodbye? ;-)

Comment (angelozerr) [reply qute.jdt/com.redhat.qute.jdt.test/projects/maven/qute-quickstart/src/main/java/org/acme/qute/HelloResource.java]: Ooops thanks :) fixed.

**Comment by angelozerr** [comment]
> CI tests fail

fixed.

Now I must clean some code and improve some performance to retrieve referenced field (today I'm searching fieldreference from Java sources folder instead of searching in teh given Java class directly). The only case that I must search in sources folder is when CheckedTemplate is used as class directly (not with inner class).

**Comment by angelozerr** [comment]
The PR is ready now.

**Comment by fbricon** [comment]
@angelozerr thanks!

**Pull request merged**